apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs

```diff
@@ -125,7 +137,7 @@ private SpannerDbType(TypeCode typeCode, int? size = null)
         public int? Size { get; }
 
         private SpannerDbType(TypeCode typeCode, SpannerDbType arrayElementType)
-            : this(typeCode) => ArrayElementType = arrayElementType;
+            : this(typeCode, arrayElementType.TypeAnnotationCode) => ArrayElementType = arrayElementType;
```

Comment:
Apart from PgNumeric, TypeAnnotationCode of all other types would be Unspecified. Only in case of array of PgNumeric being constructed that TypeAnotationCode.PgNumeric would be passed. In all other cases, Unspecified annotation would be passed.